src/Microsoft.Tye.Core/ApplicationFactory.cs

```diff
@@ -98,6 +98,13 @@ public static async Task<ApplicationBuilder> CreateAsync(OutputContext output, F
                         {
                             project.BuildProperties.Add(buildProperty.Name, buildProperty.Value);
                         }
+
+                        if (framework != null)
+                        {
+                            // Only use the TargetFramework for the "--framework" if it's not defined already from the YAML
```

Comment:
Is this comment accurate? I believe the cli options wins over yaml.

Reply:
oOo I forgot that one, I think it's a "not funny" one ╚(•⌂•)╝
It does seems obvious that a "CLI args" would "win" overall 

if `tye.yaml` have 2 projects
  * 1st in csproj use single `TargetFramework>net5.0`
  * 2nd in csproj use multi `TargetFrameworks>net5.0;netcoreapp3.1`
  * what should do `tye run --framework netcoreapp3.1` ? this would throw, and I'm fine with that, just want to be sure


I'll have to debug that again (was a long time ago), I think the was the `string? framework` is computed there was no way to differentiate if the value was passed down from the `yaml` or from the `CLI` especially when the scenario was the previous one

This lead me to also write the `e2e` you asked about :
https://github.com/dotnet/tye/pull/567#discussion_r481923444

I took the approach of the `null check` since that was the only info I had at this very place in code.
I'll try to re-debug it again to be sure of all of this, If you have any change suggestion on that one it will be more than welcome

Reply:
Hmmm this brings up a good question, if someone specifies --framework, should it apply to all projects or just projects that are multi-targeted? I find it kind of strange that if you run `tye run --framework netcoreapp3.1` you might still end up with services running on `net5.0` but I guess that is probably the most convenient. The alternative is to force the user to specify the TFM in tye.yaml instead, which is okay as well but makes it really not much different than just picking a particular TFM in the csproj directly (i.e. why bother with TargetFrameworks in the first place).

What do you think @jkotalik @glennc ?

Reply:
there's also this :
```xml
<TargetFrameworks>netcoreapp2.1;netcoreapp3.1</TargetFrameworks>
```

`tye run --framework net5.0`

if you try this with `dotnet run` IIRC, this will fail

FUNNY NOTE:
```xml
<TargetFramework>net5.0</TargetFramework>
<TargetFrameworks>netcoreapp2.1;netcoreapp3.1</TargetFrameworks>
```
This is why this `if` is here

how should these behave ?
`try run`
`tye run --framework net5.0`
`tye run --framework netcoreapp3.1`
`tye run --framework net6.0`

Reply:
for info:
not declared look like a failure today
![image](https://user-images.githubusercontent.com/2266487/92023525-2f540b80-ed5d-11ea-8a15-87a94df6e6b6.png)

Reply:
but this looks "okish"
![image](https://user-images.githubusercontent.com/2266487/92023807-9671c000-ed5d-11ea-879b-7a2516202e64.png)

Reply:
That being said and to be consistent I could try to change and force the framework everywhere and make it fails it there's no match
that's totally fine to either "force all" from CLI or specifics goes in YAML
that will avoid "magic/question"

will start to work on that just in case

Reply:
A few observations here:

1. Again, the scenario where TargetFramework and TargetFrameworks are both specified is not really valid. Don't dwell on making this "work". Let me know if you want me to elaborate on this further and explain why some of the scenarios you mentioned above "worked" and some "didn't work". It has to do with how the three steps of restore, build and run handle TargetFramework and TargetFrameworks.
2. I think it makes sense that the option `--framework` should only apply to multi-targeted apps only. This magic seems reasonable to me, and the alternative of forcing every service obey `--framework` seems even more troublesome. I'm okay with this behaviour as long as we add a test for it and it's more clear in the option description.

Reply:
for `2.` i'll wait for @jkotalik and @glennc answer so we got all the input before going wild on the branch

personal opinion:
current implementation was the easiest at the moment I did the PR and also seems logic for this very specific scenario
But after a bit of time I'm wondering if it should not just apply to every single project.
Why ?
well this would be kinda like `dotnet build` or `dotnet sln add` will throw if there's a conflict where the tools would solve that "conflict" on it own with an arbitrary logic.

I'll wait for a consensus and see if I got time to prepare the code for the "apply all" just in case

Reply:
Ah I forgot to update this thread but we met last week and agreed that the proposal "the option --framework should only apply to multi-targeted apps only" is preferred, so let's continue with that in mind.

Reply:
Does the logic here need to be updated to ensure we only override the TargetFramework when it's a multi-targeted app?